src/TraceEvent/Computers/StartStopActivityComputer.cs

```diff
@@ -957,53 +962,117 @@ void FixAndProcessAdoNetEvents(TraceEvent data)
+
+                            if (0 < data.PayloadNames.Length)
                             {
-                                if (!first)
-                                    sb.Append(' ');
-                                sb.Append(key).Append("=").Append('"').Append(valueStr).Append('"');
+                                var args = data.PayloadByName("Arguments") as DynamicTraceEventData.StructValue[];
+                                if (args != null)
+                                {
+                                    var sb = Utilities.StringBuilderCache.Acquire(64);
+                                    bool first = true;
+                                    foreach (var arg in args)
+                                    {
+                                        string key = arg["Key"] as string;
+                                        object value = arg["Value"] as string;
+                                        if (key != null && value != null)
+                                        {
+                                            string valueStr = value.ToString();
+                                            if (!string.IsNullOrEmpty(valueStr))
+                                            {
+                                                if (!first)
+                                                    sb.Append(' ');
+                                                sb.Append(key).Append("=").Append('"').Append(valueStr).Append('"');
+                                            }
+                                        }
+                                        first = false;
+                                    }
+                                    extraInfo = Utilities.StringBuilderCache.GetStringAndRelease(sb);
+                                }
                             }
-                        }
-                        first = false;
+                            OnStart(data, extraInfo, null, null, null, taskName);
+                            break;
+                        case "Microsoft.AspNetCore.Hosting.BeginRequest":
+                            // Just change the name from "Activity1Start" to "AspNetCoreHosting".
+                            OnStart(data, null, null, null, null, "AspNetCoreHosting");
+                            break;
                     }
-                    extraInfo = Utilities.StringBuilderCache.GetStringAndRelease(sb);
+                    result = true;
                 }
-                OnStart(data, extraInfo, null, null, null, taskName);
-                return true;
             }
             catch (Exception ex)
             {
                 Trace.WriteLine("Threw exception while processing DiagnosticSource start events: " + ex.Message);
             }
-            return false;
+
+            return result;
+        }
+
+        void FixAndProcessAppInsightsEvents(TraceEvent data)
```

Comment:
I am OK with adding these for the ones we know about, but this is clearly a place where we should have an extensiblity point since client will care about ones we have not done, or want something different.   We can think about this.

Reply:
Are you saying this is a good point to allow user providing sort of pre-processors for the TraceEvents? That I agree.